pycbc/inject/injfilterrejector.py

```diff
@@ -231,6 +259,95 @@ def from_cli_multi_ifos(cls, opt, ifos):
             inj_filter_rejectors[ifo] = cls.from_cli_single_ifo(opt, ifo)
         return inj_filter_rejectors
 
+    def get_inj_end_times(self):
+        """Return a list of the injection end times."""
+        if self._end_times is None:
+            self._end_times = []
+            for inj in self.injection_params.table:
+                if isinstance(inj, np.record):
+                    # hdf format file
+                    end_time = inj['tc']
+                else:
+                    # must be an xml file originally
+                    end_time = inj.geocent_end_time + \
+                        1E-9 * inj.geocent_end_time_ns
+                self._end_times.append(end_time)
+            self._end_times = np.array(self._end_times)
+        return self._end_times
+
+    # Written together with Google Gemini
+    def precompute_injection_intervals(self):
+        """
+        Precompute and merges windows around the injections to speed up
+        determining if triggers are in injection times later.
+        """
+        window = self.inj_trigger_window
+        inj_times = self.get_inj_end_times()
+        if inj_times.size == 0:
+            return np.empty((0, 2))
+
+        # Create individual intervals
+        starts = inj_times - window
+        ends = inj_times + window
+        intervals = np.c_[starts, ends]
+
+        # Sort intervals by their start points
+        intervals = intervals[intervals[:, 0].argsort()]
```

Comment:
Can you not simply sort `inj_times` beforehand (maybe even in `get_inj_end_times()`, so you cache them already sorted)?